### PR TITLE
Update Cirrus macOS environment to M1/Ventura

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
LGTM

File: .cirrus.yml [hunk 1]
```diff
@@ -40,15 +40,15 @@ task:
...
+    image: ghcr.io/cirruslabs/macos-ventura-base:latest
 
 task:
   name: tests-macos-1
   use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_PR == ''
   script: scripts/verify_tests_on_master.sh --shards 2 --shard-index 1
-  osx_instance:
-    image: mojave-base
+  macos_instance:
+    image: ghcr.io/cirruslabs/macos-ventura-base:latest
```
Comment (jmagman) [inline .cirrus.yml:51]:
```suggestion
    image: ghcr.io/cirruslabs/macos-ventura-xcode:14
```

File: .cirrus.yml [hunk 2]
```diff
@@ -40,15 +40,15 @@ task:
   name: tests-macos-0
   use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_PR == ''
   script: scripts/verify_tests_on_master.sh --shards 2 --shard-index 0
-  osx_instance:
-    image: mojave-base
+  macos_instance:
+    image: ghcr.io/cirruslabs/macos-ventura-base:latest
```
Comment (jmagman) [inline .cirrus.yml:44]: This should be pinned to a specific Xcode version to minimize out-of-band failures.
https://github.com/flutter/plugins/blob/320461910156084cf88aaceb7873c958b455ba7d/.cirrus.yml#L38

```suggestion
    image: ghcr.io/cirruslabs/macos-ventura-xcode:14
```

Comment (jason-simmons) [reply .cirrus.yml]: done

**Review by jmagman — APPROVED** [review]
LGTM once tests are passing

**Comment by jason-simmons** [comment]
The change to the Cirrus `macos-ventura-xcode` image failed because that image installs a version of Flutter and adds it to the `PATH`
(https://github.com/cirruslabs/macos-image-templates/blob/master/templates/xcode.pkr.hcl#L95)

**Comment by jmagman** [comment]
> The change to the Cirrus `macos-ventura-xcode` image failed because that image installs a version of Flutter and adds it to the `PATH` (https://github.com/cirruslabs/macos-image-templates/blob/master/templates/xcode.pkr.hcl#L95)

Hm, maybe need to make a `mac_verify_tests_on_master.sh` version that doesn't clone Flutter but instead checks out the master channel?
(`verify_tests_on_master` is used on Linux too)
https://github.com/flutter/tests/blob/main/scripts/verify_tests_on_master.sh

https://github.com/flutter/plugins/blob/320461910156084cf88aaceb7873c958b455ba7d/.cirrus.yml#L50-L54

File: .cirrus.yml [hunk 3]
```diff
@@ -40,15 +40,15 @@ task:
   name: tests-macos-0
   use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_PR == ''
   script: scripts/verify_tests_on_master.sh --shards 2 --shard-index 0
-  osx_instance:
-    image: mojave-base
+  macos_instance:
+    image: ghcr.io/cirruslabs/macos-ventura-xcode:14
```
Comment (jmagman) [inline .cirrus.yml:44]: Or: does `macos-ventura-vanilla:13.1` work? Not sure if this repo needs `brew` installed.
https://github.com/cirruslabs/macos-image-templates/pkgs/container/macos-ventura-vanilla

Comment (jason-simmons) [reply .cirrus.yml:44]: `macos-ventura-vanilla:13.1` failed with a different error:

`xcode-select: note: no developer tools were found at '/Applications/Xcode.app', requesting install. Choose an option in the dialog to download the command line developer tools.`

(https://cirrus-ci.com/task/5306558125441024?logs=main#L17)

**Comment by jason-simmons** [comment]
Updated this to use the `macos-ventura-xcode:14` image and to move the Flutter master branch to the head of the `PATH` when running the tests.

The tests run to completion on Cirrus with those changes.

But the DanTup_tiler test is failing some golden image checks when running in the Cirrus M1 setup.

**Comment by DanTup** [comment]
FYI - I've included the changes here in my PR to bump the tiler tests (since both are required to have green bots):

https://github.com/flutter/tests/pull/214

**Comment by DanTup** [comment]
https://github.com/flutter/tests/pull/214 has landed now.

**Pull request closed without merging**